health/state.go

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package health
+
+import (
+	"time"
+)
+
+// State is a representation of the health status of the backend, which is
+// provided to the client UI via LocalAPI. It is used to simplify the data structure
+// provided over JSON.
+type State struct {
+	// Warnings maps each WarnableCode to a WarningStateRepresentation.
+	Warnings map[WarnableCode]WarningStateRepresentation
+}
+
+type WarningStateRepresentation struct {
+	WarnableCode WarnableCode
+	Severity     Severity
+	Title        string
+	Text         string
+	BrokenSince  time.Time
+	Args         map[string]string
```

Comment:
omitempty on these? But then this needs to be *time.Time because omitempty doesn't work on zero structs like time.Time.

Reply:
done